tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,9 +876,55 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
-}
 
-class MyFetchClass
-{
-    public $test_int, $test_string, $test_datetime;
+    public function getConnections()
+    {
+        return [
+            [MysqliConnection::class],
```

Comment:
If we extract the iteration logic into a separate class, we shouldn't be testing the connection/statement classes anymore. This can be moved out to a brand new `StatementIteratorTest` class and tested with a `Statement` mock. With this approach, we'll not need to instantiate connection/statement classes and deal with their runtime dependencies.

Reply:
Ok, did some tidy up, moved tests, had to change the mock a little bit, the yield didn't quite work. Cheers for your help.